build/go-version-check.sh

```diff
@@ -7,7 +7,7 @@
 
 required_version_major=1
 minimum_version_minor=15
-minimum_version_15_patch=3
+minimum_version_15_patch=14
```

Comment:
Can we set this to 11 instead of 14? 

This is to follow [this commit](https://github.com/cockroachdb/cockroach/commit/e184d516149399d74804395407f624a9eca46f97#diff-7f72b2b2a09ada0b8f90bc523f06e0f871bbeb25d9ebba61464917c9fe9a7a55) and [issue](https://github.com/cockroachdb/cockroach/issues/63836).

Reply:
Sure, let me change it